### PR TITLE
SAH-48

### PR DESCRIPTION
**Pull request by eng-atwijukire**
## Tasks

- [x] Docker files for production-ready images for all apps: website, API, agent, client
- [x] Docker compose file to build & run images locally
- [x] Local dev config for Hasura engine & API app dependencies (db & redis)
- [x] CI Build test for API

## Reasons
- We need to run all apps in docker containers providing an advantage of portability, scalability and other more benefits

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -13,19 +13,129 @@ services:
...
       context: .
       dockerfile: ./infra/docker/Dockerfile.api
     image: sahil-api
     restart: always
-    depends_on: [database]
     ports:
-      - '3001:3001'
+      - '8000:8000'
+    env_file:
+      - .env
```
Comment (Emmanuel-Melon) [inline docker-compose.yml]: It's currently directed to the .env file at the root, which doesn't have any relevant secrets. It should be directed to the .env file inside the API.

Comment (eng-atwijukire) [reply docker-compose.yml]: Yes had fixed that but forgot to push the changes

**Comment by eng-atwijukire** [comment]
@Emmanuel-Melon I have updated your recommandations

**Review by Emmanuel-Melon — APPROVED** [review]
LGTM

**Pull request merged**